src/main/java/me/zeroeightsix/kami/gui/rgui/Component.kt

```diff
@@ -0,0 +1,126 @@
+package me.zeroeightsix.kami.gui.rgui
+
+import me.zeroeightsix.kami.KamiMod
+import me.zeroeightsix.kami.module.modules.client.ClickGUI
+import me.zeroeightsix.kami.setting.GuiConfig.setting
+import me.zeroeightsix.kami.util.Wrapper
+import me.zeroeightsix.kami.util.graphics.VertexHelper
+import me.zeroeightsix.kami.util.graphics.font.HAlign
+import me.zeroeightsix.kami.util.graphics.font.VAlign
+import me.zeroeightsix.kami.util.math.Vec2f
+
+open class Component(
+    name: String,
+    posXIn: Float,
+    posYIn: Float,
+    widthIn: Float,
+    heightIn: Float,
+    val settingGroup: SettingGroup
+) {
+
+    // Basic info
+    val originalName = name
+    val name = setting("Name", name, { false })
+    val visible = setting("Visible", true, { false }, { _, it -> it || !closeable })
+
+    protected val relativePosX = setting("PosX", posXIn, -69420.911f..69420.911f, 0.1f, { false },
+        { _, it -> if (this is WindowComponent && KamiMod.isInitialized()) absToRelativeX(relativeToAbsX(it).coerceIn(2.0f, scaledWidth - width.value - 2.0f)) else it })
+    protected val relativePosY = setting("PosY", posYIn, -69420.911f..69420.911f, 0.1f, { false },
+        { _, it -> if (this is WindowComponent && KamiMod.isInitialized()) absToRelativeY(relativeToAbsY(it).coerceIn(2.0f, scaledHeight - height.value - 2.0f)) else it })
+
```

Comment:
This causes a crash when height of the component is more than 4 pixels less than the height of the window, same with the horizontal. This is caused by the coerceIn() having a higher maximum than minimum at these times.

Reply:
Fixed already. ty